File: Cpp/977_Squares_of_a_Sorted_Array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> result(nums.size());
        int left = 0;
        int right = nums.size() - 1;
        
        for(int i = nums.size() - 1; i >= 0; --i){
            if(abs(nums[left]) < abs(nums[right])) {
                result[i] = nums[right] * nums[right--];
            } else {
                result[i] = nums[left] * nums[left++];
            }
        }
        
        return result;
    }
};

File: Kotlin/977_Squares_of_a_Sorted_Array.kt
class Solution {
    fun sortedSquares(nums: IntArray): IntArray {
        val result = IntArray(nums.size)
        var left = 0
        var right = nums.size - 1
        
        for(i in nums.size - 1 downTo 0){
            if(abs(nums[left]) < abs(nums[right])) {
                result[i] = nums[right] * nums[right--]
            } else {
                result[i] = nums[left] * nums[left++]
            }
        }
        
        return result
    }
}